pkg/translate/file.go

```diff
@@ -49,70 +52,116 @@ func (u *uploadParams) validate() error {
 		return fmt.Errorf("'schema' is required")
 	}
 
+	if u.serviceID == uuid.Nil {
+		return fmt.Errorf("'service_id' is required")
+	}
+
+	if u.languageTag == language.Und {
+		return fmt.Errorf("'language' is required")
+	}
+
 	return nil
 }
 
 func (t *TranslateServiceServer) UploadTranslationFile(
 	ctx context.Context,
 	req *translatev1.UploadTranslationFileRequest,
 ) (*emptypb.Empty, error) {
-	uploadReq := (*uploadTranslationFileRequest)(req)
-
-	params, err := uploadReq.parseParams()
+	params, err := parseUploadTranslationFileRequestParams(req)
 	if err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	if err := params.validate(); err != nil {
+	if err = params.validate(); err != nil {
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	return &emptypb.Empty{}, nil
+	messages, err := MessagesFromData(params.schema, params.data)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	// Some converts do not provide language, so we override it with one from request for consistency.
+	messages.Language = params.languageTag
```

Comment:
For now, a temporary solution. I will change it when working on this ticket [
Detect language from uploaded file body](https://github.com/orgs/translate-agent/projects/1?pane=issue&itemId=25630671)

Reply:
Fixed on #64